packages/python/plotly/plotly/tests/test_optional/test_px/test_px_hover.py

```diff
@@ -183,8 +189,15 @@ def test_sunburst_hoverdict_color():
     assert "color" in fig.data[0].hovertemplate
 
 
-def test_date_in_hover():
-    df = pd.DataFrame({"date": ["2015-04-04 19:31:30+1:00"], "value": [3]})
-    df["date"] = pd.to_datetime(df["date"])
-    fig = px.scatter(df, x="value", y="value", hover_data=["date"])
-    assert str(fig.data[0].customdata[0][0]) == str(df["date"][0])
+def test_date_in_hover(request, constructor):
```

Comment:
@FBruzzesi What does this mean in practice, in terms of how we should document this limitation in the docs? Something like "Datetimes converted to strings for non-Pandas input dataframes will lose their timezone information"?

Reply:
ooh this is a really good point, thanks! I just tried this out, and indeed, there is a difference:
- for pandas, the local datetime is displayed
- for non-pandas, the UTC datetime is displayed

@FBruzzesi here's an example to test this with:

```python
df = pl.DataFrame({'price': [40, 50], 'datetime': [datetime(2020,1,1), datetime(2020,1,2)]})
df = df.with_columns(pl.col('datetime').dt.replace_time_zone('Asia/Kathmandu'))
px.scatter(df, x='datetime', y='price')
```

then, for Polars, we see:

![image](https://github.com/user-attachments/assets/d925fc6c-289e-4389-81a7-9ae26817df37)


whereas for pandas:

![image](https://github.com/user-attachments/assets/bd299812-e5f3-40ed-8481-264bd07acb02)

---

I'll have to check the logic, but the way to get the local datetime in Narwhals would be `.dt.replace_time_zone(None)`